codes/quantum/qubits/subsystem/subsystem-gen-shor.yml

```diff
@@ -0,0 +1,50 @@
+
+  \begin{align*}
+  S_i &= \otimes_{j=1}^{n_1}Z^{(H_1)_{ij}} & T_i &= \otimes_{j=1}^{n_2}X^{(H_2)_{ij}} \\
+  \mathcal{S}_1 &=\langle S_1,\ldots,S_{n_1-k_1}\rangle & \mathcal{S}_2 &= \langle T_1,\ldots,T_{n_2-k_2} \rangle
+  \end{align*}
+  
+  Let \(n = n_1 + n_2\). Consider a square lattice of size \(n \times n\) with a qubit located at each vertex. Let \(\mathcal{T}_1\) be the stabilizer group made up of letting \(\mathcal{S}_1\) operators acting on all columns of the lattice. Let \(\mathcal{T}_2\) be the stabilizer group made up of letting \(\mathcal{S}_2\) operators acting on all rows of the lattice. Then \(\mathcal{T} = \langle\mathcal{T}_1,\mathcal{T}_2\rangle\). \(\mathcal{T}\) is non-abelian and from which we can construct an abelian invariant subgroup \(\mathcal{S}\) of \(\mathcal{T}\). \(\mathcal{S}\) is the group generated by \(\mathcal{S}_1\) and \(\mathcal{S}_2\) for all stabilizer codeword combinations in both rows and columns. By construction, \(\mathcal{S}\) is a stabilizer with \((n_1 - k_1)k_2\) stabilizer generators (i.e., the tensor product of \(I\) and \(Z\) operators), and  \((n_2 - k_2)k_1\) stabilizer generators (i.e., the tensor product of \(I\) and \(X\) operators). Next, we can construct a group of logical Paulis on \(k_1k_2\) qubits using the logical operators of \(\mathcal{S}_1\) and \(\mathcal{S}_2\).
+
+  Based on the above construction, the Hilbert space on \(n_1n_2\) qubits can be decomposed into a multiple direct sums of multiple tensor products of Hilbert spaces of lower dimensions, as outlined in \cite{arXiv:quant-ph/0506023}. Then the error recovery procedure could correct information encoded into the subsystem.
+
+    
+protection: 'Detecting single-qubit errors of weight \(min(d_1,d_2)\), this code corrects bit flip and phase flip errors modulo the subsystem structure.'
+
+features:
+    
+  decoders:
+    - 'By measuring the generators of \(\mathcal{S}\), we can correct errors related to the correctable errors of the classical linear codes.'
+    - 'In a generalized \([[n1n2, k1k2, min(d1, d2)]]\) Shor code, the error correction is achieved by measuring (n1−k1)n2+(n2−k2) stabilizer generators \cite{arXiv:quant-ph/0508131}. This subsystem generalized Shor code has achieved the same parameters for the code but using only \((n1−k1)k2+k1(n2−k2)\) stabilizer measurements.'
+
+realizations:
+  - 'Realized in all-photonic quantum repeater architecture \cite{arxiv:2203.07979}.'
+
+
+relations:
+  parents:
```

Comment:
@myself, child is quantum_parity; cousin is bacon_shor; remove cousin from quantum_parity; add cousin to hypergraph_product, see Tillich Zemor paper pg. 18, include that description